src/peft/peft_model.py

```diff
@@ -288,6 +288,36 @@ def forward(self, *args, **kwargs):
         else:
             return self.base_model.model(*args, **kwargs)
 
+    def prepare_model_for_training(self):
+        r"""
+        This method wrapps the entire protocol for preparing a model before running a training. This includes:
+            1- Cast the layernorm in fp32 2- making output embedding layer require grads
+        """
+        loaded_in_8bit = getattr(self.base_model, "is_loaded_in_8bit", False)
+
+        for param in self.base_model.parameters():
+            # freeze base model's layers
+            param.requires_grad = False
```

Comment:
this can't be called after creating PeftModel as adapter weights are added and are trainable but with this they will be frozen. LoRA weights are injected in base model inplace.